references/cuegen/types.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cuegen
+
+import (
+	"fmt"
+	goast "go/ast"
+	gotypes "go/types"
+
+	"golang.org/x/tools/go/packages"
+)
+
+type typeInfo map[gotypes.Type]*goast.StructType
+
+func getTypeInfo(p *packages.Package) typeInfo {
+	m := make(typeInfo)
+
+	for _, f := range p.Syntax {
+		goast.Inspect(f, func(n goast.Node) bool {
+			// record all struct types
+			if t, ok := n.(*goast.StructType); !ok {
+				m[p.TypesInfo.TypeOf(t)] = t
```

Comment:
What will happen if `t` is not *goast.StructType, it will not be stored in `m`? Maybe some tests will make this easier to understand.

Reply:
Fixed